modules/account-lib/src/coin/algo/transactionBuilder.ts

```diff
@@ -1,29 +1,301 @@
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/
+   */
+  lastRound(round: number): this {
+    this.validateValue(new BigNumber(round));
+
+    this._lastRound = round;
+
+    return this;
+  }
+
+  /**
+   * Sets the lease on the transaction.
+   *
+   * A lease is a mutex on the transaction that prevents any other transaction
+   * from being sent with the same lease until the prior transaction's last
+   * round has passed.
+   *
+   * @param {Uint8Array} lease The lease to put the transaction.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/
+   */
+  lease(lease: Uint8Array): this {
+    this._lease = lease;
+
+    return this;
+  }
+
+  /**
+   * Sets the note for the transaction.
+   *
+   * @param {Uint8Array} note Arbitrary data for sender to store.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/
+   */
+  note(note: Uint8Array): this {
+    this._note = note;
+
+    return this;
+  }
+
+  /**
+   * Sets the authorized address.
+   *
+   * The authorized asset will be used to authorize all future transactions.
+   *
+   * @param {BaseAddress} authorizer The address to delegate authorization authority to.
+   * @returns {TransactionBuilder} This transaction builder.
+   *
+   * @see https://developer.algorand.org/docs/reference/transactions/
+   */
+  reKeyTo(authorizer: BaseAddress): this {
+    this.validateAddress(authorizer);
+    this._reKeyTo = authorizer.address;
+
+    return this;
   }
 
   /** @inheritdoc */
   validateAddress({ address }: BaseAddress): void {
-    throw new NotImplementedError('validateAddress not implemented');
+    if (!algosdk.isValidAddress(address)) {
+      throw new AddressValidationError(address);
+    }
+  }
+
+  /** @inheritdoc */
+  protected async buildImplementation(): Promise<Transaction> {
+    this.transaction.signMultiSig(this._keyPairs);
+
+    return this._transaction;
   }
 
   /** @inheritdoc */
+  protected fromImplementation(rawTransaction: Uint8Array | string): Transaction {
+    const buffer = typeof rawTransaction === 'string' ? Buffer.from(rawTransaction, 'hex') : rawTransaction;
+    const algosdkTxn: algosdk.Transaction = algosdk.decodeUnsignedTransaction(buffer);
+
+    this._fee = algosdkTxn.fee;
+    this._sender = algosdk.encodeAddress(algosdkTxn.from.publicKey);
+    this._genesisHash = algosdkTxn.genesisHash.toString('base64');
+    this._genesisId = algosdkTxn.genesisID;
+    this._firstRound = algosdkTxn.firstRound;
+    this._lastRound = algosdkTxn.lastRound;
+    this._lease = algosdkTxn.lease;
+    this._note = algosdkTxn.note;
+    this._reKeyTo = algosdkTxn.reKeyTo ? algosdk.encodeAddress(algosdkTxn.reKeyTo.publicKey) : undefined;
+
+    this._transaction = new Transaction(this._coinConfig);
```

Comment:
`_transaction` has been initialized in the constructor. We need to set algoTransaction in that obj.